### PR TITLE
Add policy for private DNS zone groups configuration for Static Web Apps

### PR DESCRIPTION
**Pull request by digitmj**

### CODE REVIEW

**Review by Joshua-Donovan — CHANGES_REQUESTED** [review]
A couple of small changes are needed before we can approve this pull request, please let me know if you have questions.

File: policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    "mode": "Indexed",
+    "parameters": {
+      "privateDnsZoneId": {
+        "type": "String",
+        "metadata": {
+          "displayName": "Private DNS zone",
+          "description": "Specifies the private DNS zone to use to configure private endpoint. Make sure to have a dedicated DNS zone for each possible partition id, i.e. 'privatelink.2.azurestaticapps.net' for partition id 2 and so on.",
+          "strongType": "Microsoft.Network/privateDnsZones"
+        },
+        "defaultValue": "/subscriptions/2d028fdd-9fba-467a-8180-722e8ac4d690/resourceGroups/sharedresources/providers/Microsoft.Network/privateDnsZones/privatelink.azurestaticapps.net"
```
[the remark this answers is not in the document]
Comment (digitmj) [reply policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.json]: Understandable, I've removed the defaultValue here.

File: policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "privateDnsZoneId": {
+    "type": "String",
+    "metadata": {
+      "displayName": "Private DNS zone",
+      "description": "Specifies the private DNS zone to use to configure private endpoint. Make sure to have a dedicated DNS zone for each possible partition id, i.e. 'privatelink.2.azurestaticapps.net' for partition id 2 and so on.",
+      "strongType": "Microsoft.Network/privateDnsZones"
+    },
+    "defaultValue": "/subscriptions/2d028fdd-9fba-467a-8180-722e8ac4d690/resourceGroups/sharedresources/providers/Microsoft.Network/privateDnsZones/privatelink.azurestaticapps.net"
```
[the remark this answers is not in the document]
Comment (digitmj) [reply policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.parameters.json]: Understandable, I've removed the defaultValue here.

File: policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          }
+        ]
+      },
+      "then": {
+        "effect": "[parameters('effect')]",
+        "details": {
+          "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+          "evaluationDelay": "AfterProvisioning",
+          "roleDefinitionIds": [
+            "/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7"
```
[the remark this answers is not in the document]
Comment (digitmj) [reply policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.json]: In my testing environment it seems to work somehow, but you're right, Reader RBAC role isn't sufficient. Updated the guid to the guid of Network Contributor.

File: policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.rules.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    ]
+  },
+  "then": {
+    "effect": "[parameters('effect')]",
+    "details": {
+      "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+      "evaluationDelay": "AfterProvisioning",
+      "roleDefinitionIds": [
+        "/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7"
```
[the remark this answers is not in the document]
Comment (digitmj) [reply policyDefinitions/App Service/configure-a-private-dns-zone-for-static-webapp/azurepolicy.rules.json]: In my testing environment it seems to work somehow, but you're right, Reader RBAC role isn't sufficient. Updated the guid to the guid of Network Contributor.

**Comment by Joshua-Donovan** [comment]
Thank you @digitmj for updating the policy, the team will review for final pull request approval. Have a great day!

**Review by Joshua-Donovan — APPROVED** [review]

**Pull request merged**